### PR TITLE
Allow private methods to inherit default annotations from package or class scope

### LINKED ISSUE #374: @ParametersAreNonnullByDefault ignored for private methods if set on package or class level.
We have a problem here where spotbugs does not apply the "@ParametersAreNonnullByDefault" annotation to private methods if the annotation has been set on package or class level. If you annotate the private method directly however it works.

How to reproduce:
1) Unzip the attached maven project.
2) Run "mvn clean verify"

Expected Behavior:
Spotbugs should find the same bug in all three example classes, as all classes have the same code, but simply get their @ParametersAreNonnullByDefault annotation from a different level (package/class/method).

Actual Behavior:
Spotbugs only finds the bug in "TestNonnullByDefaultFromMethodAnnotation.java"

Workaround:
Change all private methods to protected or public and spotbugs will correctly complain about all three classes.

[spotbugs.zip](https://github.com/spotbugs/spotbugs/files/1298881/spotbugs.zip)

### PR DESCRIPTION
**Pull request by cwbriones**
Resolves #374.

Thanks @RitChan for the reproduction test cases attached to the Issue.

From the comment in the code, the original decision to omit private methods seems very much intentional. But since the issue is classified as a bug it seemed safe to just remove that check.

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue374Test.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Contributions to SpotBugs
+ * Copyright (C) 2020, kengo
```
Comment (KengoTODA) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue374Test.java:3]: Use your name, please :)

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/jsr305/TypeQualifierApplications.java [hunk 2]
```diff
@@ -986,11 +986,6 @@ private static TypeQualifierAnnotation computeEffectiveTypeQualifierAnnotation(T
             // default annotations
         }
 
-        /** private methods don't inherit from class or package scope */
-        if (xmethod.isPrivate()) {
-            stopAtMethodScope = true;
-        }
-
```
Comment (KengoTODA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/ba/jsr305/TypeQualifierApplications.java:993]: I'm still not sure about the intention behind this code. 
But commit message of b1fcbae0c11852eb42e5e2f7e1eca61003aa4a7b tells nothing, so I agree to change the behavior to make it intuitive.

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@ Currently the versioning policy of this project follows [Semantic Versioning v2.
 
 ### Fixed
 * Fixed not working detector 'CbeckMustOverrideSuperAnnotation' and renamed to 'OverridingMethodsMustInvokeSuperDetector'
+* Allow private methods to inherit default annotations from package or class scope. ([#374](https://github.com/spotbugs/spotbugs/issues/374))
```
Comment (KengoTODA) [inline CHANGELOG.md:13]: Please merge the latest `origin/master` to your branch, and move your entry to under `## Unreleased`.

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue374Test.java (hunk 1 above)
Comment (RitChan) [reply spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue374Test.java:3]: I just copied from another place and didn't notice that XD. 
Can you correct this copyright owner in your repository @cwbriones? You can use "ritchan". Thanks!

**Pull request merged**